### PR TITLE
BigQuery Storage: Update quickstart sample with data format and sharding options.

### PR DESCRIPTION
**Pull request by tswast**
* Moves sample to `samples/` directory.
* Updates `synth.py` to use common templates (which now includes samples tests in the noxfile).

### CODE REVIEW

**Comment by tswast** [comment]
Comments copied from @shollyman's https://github.com/GoogleCloudPlatform/golang-samples/pull/906.

File: bigquery_storage/setup.cfg [hunk 1]
```diff
@@ -1,10 +1,3 @@
...
 [bdist_wheel]
 universal = 1
-
-[flake8]
-exclude =
-    *_pb2.py
-    *_pb2_grpc.py
-    google/cloud/bigquery_storage_v1beta1/gapic/big_query_storage_client.py
-    google/cloud/bigquery_storage_v1beta1/gapic/transports/big_query_storage_grpc_transport.py
-    tests/unit/gapic/v1beta1/test_big_query_storage_client_v1beta1.py
```
Comment (tseaver) [inline bigquery_storage/setup.cfg]: Is this change intentional?  Seems like it might break stuff.

File: bigquery_storage/synth.py [hunk 2]
```diff
@@ -130,4 +130,10 @@
 )
 # END: Ignore lint and coverage
 
+# ----------------------------------------------------------------------------
+# Add templated files
+# ----------------------------------------------------------------------------
+templated_files = common.py_library(unit_cov_level=79, cov_level=79, samples_test=True, system_test_dependencies=["fastavro", "pandas", "pyarrow != 0.14.0"])
+s.move(templated_files)
```
Comment (tseaver) [inline bigquery_storage/synth.py]: Might need one or more excludes here.

File: bigquery_storage/setup.cfg (hunk 1 above)
Comment (tswast) [reply bigquery_storage/setup.cfg]: Linter is passing. Seems these lines are no longer needed.

File: bigquery_storage/synth.py (hunk 2 above)
Comment (tswast) [reply bigquery_storage/synth.py]: I'd rather keep the common templates if possible. It seems to be working.
